knowndevices.go

```diff
@@ -1,21 +1,26 @@
 package broadlinkgo
 
 type deviceCharacteristics struct {
-	known     bool
-	name      string
-	supported bool
-	ir        bool
-	rf        bool
-	power     bool
+	deviceType        int
+	known             bool
+	name              string
+	supported         bool
+	ir                bool
+	rf                bool
+	power             bool
+	requestHeader     []byte
+	codeSendingHeader []byte
 }
 
 type knownDevice struct {
-	deviceType int
-	name       string
-	supported  bool
-	ir         bool
-	rf         bool
-	power      bool
+	deviceType        int
+	name              string
+	supported         bool
+	ir                bool
+	rf                bool
+	power             bool
+	requestHeader     []byte
+	codeSendingHeader []byte
```

Comment:
Instead of adding the explicit bytes (which are the same for all RM4 devices), I would simply add a flag indicating whether they are using the RM4 protocol:

```suggestion
	isRM4 bool
```

Reply:
BTW, I don't really think they are headers with fixed bytes. [AFAIU from python-broadlink](https://github.com/mjg59/python-broadlink/blob/c3bb598e27d27ec1e86f8a83efa3f9a3d1887c17/broadlink/remote.py#L74-L81), the RM4 protocol includes the data length.

Reply:
Ah, yes, looks like you're right.
I actually copied the python-broadlink approach here, but sometime later they changed the fixed bytes to data length: https://github.com/mjg59/python-broadlink/pull/490

Reply:
> Instead of adding the explicit bytes (which are the same for all RM4 devices), I would simply add a flag indicating whether they are using the RM4 protocol:

Now that we now that the bytes aren't fixed, I guess that might be a better approach.
I'm afraid though that I don't have the RM4 device anymore (made the changes for a friend of mine, and afterwards returned the device to him), so I cannot test any additional changes I'm afraid :(

Reply:
Not sure I understand what you guys are discussing here but I just got an RM4 Mini and downloaded and compiled from @2opremio repository and it works beautifully... thanks a lot!